matrix_client/client.py

```diff
@@ -161,6 +161,27 @@ def login_with_password_no_sync(self, username, password):
         self.api.token = self.token
         return self.token
 
+    def login_with_jwt_no_sync(self, jwt):
+        """ Login to the homeserver using a JSON Web Token.
+
+        Args:
+            jwt (str): JWT
+
+        Returns:
+            str: Access token
+
+        Raises:
+            MatrixRequestError
+        """
+        response = self.api.login(
+            "m.login.jwt", token=jwt
+        )
+        self.user_id = response["user_id"]
+        self.token = response["access_token"]
+        self.hs = response["home_server"]
+        self.api.token = self.token
```

Comment:
Consider moving code shared between `login_with_jwt_no_sync` and
`login_with_password_no_sync` into a single helper method to follow
DRY.